### PR TITLE
[core] updating AMatrix

### PR DESCRIPTION
**Pull request by pooyan-dadvand**
This update:
- Adds fill methods: `fill(value)` and `fill_identity()`
- Fixes unary minus compile problem. (@philbucher this should solve the issue in #3450) 
- Fixes some warnings for unused variables

### CODE REVIEW

**Comment by philbucher** [comment]
many errors like this:
~~~
In file included from /home/philipp/software/Kratos_MasterBranch/kratos/utilities/exact_mortar_segmentation_utility.h:24:0,
                 from /home/philipp/software/Kratos_MasterBranch/kratos/utilities/exact_mortar_segmentation_utility.cpp:20:
/home/philipp/software/Kratos_MasterBranch/kratos/includes/mortar_classes.h: In member function ‘void Kratos::MortarKinematicVariables<TNumNodes, TNumNodesMaster>::PrintInfo(std::ostream&) const’:
/home/philipp/software/Kratos_MasterBranch/kratos/includes/mortar_classes.h:401:31: error: ambiguous overload for ‘operator<<’ (operand types are ‘std::basic_ostream<char>’ and ‘const Vector {aka const Kratos::Internals::Matrix<double, 0, 1>}’)
         rOStream << "NSlave:" << NSlave << std::endl;
~~~

**Comment by pooyan-dadvand** [comment]
in the core?

**Comment by philbucher** [comment]
yes

File: external_libraries/a_matrix/include/matrix.h [hunk 1]
```diff
@@ -263,4 +273,20 @@ inline std::ostream& operator<<(std::ostream& rOStream,
     return rOStream;
 }
 
+template <typename TExpressionType, std::size_t TCategory>
+inline std::ostream& operator<<(std::ostream& rOStream,
```
Comment (philbucher) [inline external_libraries/a_matrix/include/matrix.h:278]: it works if I comment this function

**Comment by philbucher** [comment]
@pooyan-dadvand this solves the issues in MeshMovingApp, thanks

File: external_libraries/a_matrix/include/matrix.h (hunk 1 above)
Comment (pooyan-dadvand) [reply external_libraries/a_matrix/include/matrix.h:278]: oops, seems that one commit is missing here. You may comment this function safely. 

We have another version of this function in Kratos which mimics the ublas output....

**Comment by pooyan-dadvand** [comment]
I have just added a macro to make it work in both cases. It should solve the compilation error

**Review by RiccardoRossi — APPROVED** [review]

**Pull request merged**